pkg/cstor/cstor.go

```diff
@@ -506,3 +499,35 @@ func (p *Plugin) SetVolumeID(unstructuredPV runtime.Unstructured, volumeID strin
 
 	return &unstructured.Unstructured{Object: res}, nil
 }
+
+func (p *Plugin) getVolInfo(volumeID, snapName string) (*Volume, error) {
+	// To support namespace re-mapping for cloud snapshot,
+	// change createPVC to getPVCInfo
+	fn := p.createPVC
+	if p.local {
+		fn = p.getPVInfo
+	}
+
+	vol, err := fn(volumeID, snapName)
+	if err != nil {
+		return nil, err
+	}
+
+	// To support namespace re-mapping for cloud-snapshot remove below check
+	if p.local {
+		// Let's rename PV if already created
```

Comment:
few comments on what cases it can exists and what cases it doesn't would be good to have.
With new 'p.local' conditions, I guess - more such comments are good to have in code.

Reply:
ok